plugins/repository-azure/src/main/java/org/opensearch/repositories/azure/AzureStorageService.java

```diff
@@ -159,10 +162,27 @@ public Tuple<BlobServiceClient, Supplier<Context>> client(String clientName, BiC
         return new Tuple<>(state.getClient(), () -> buildOperationContext(azureStorageSettings));
     }
 
+    public StorageEndpoint getStorageBlobEndpoint(final AzureStorageSettings settings) {
+        try {
+            String endpointSuffix = settings.getEndpointSuffix();
+            if (!Strings.hasText(endpointSuffix)) {
+                endpointSuffix = "core.windows.net";
```

Comment:
What does this look like on the 1.x port? We just hardcode the suffix?

Reply:
Yes, it is the same for 1.x

With the hardcoded the suffix, I was referencing the `fromStorageSettings` method  in`azure-storage-common-12.21.2.jar!/com/azure/storage/common/implementation/connectionstring/StorageEndpoint.class`. There are using hard coded values, So i just follow its implementation.